src/missions/use-after-free-control-flow/temporal-mission.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ * Copyright (c) 2020 Microsoft, Inc.
+ */
+#include <assert.h>
+#include <ctype.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/queue.h>
+
+#if defined(CAPREVOKE) && __CHERI_PURE_CAPABILITY__
+#include <sys/caprevoke.h>
+static void __attribute__((used)) *check_caprevoke = caprevoke;
+extern void malloc_revoke(void);
+static void __attribute__((used)) *check_malloc_revoke = malloc_revoke;
+#endif
+
+struct farm;
+struct crop;
+
+static void
+success(struct farm *f)
+{
+	fprintf(stderr, "Exploit successful: computer bought the farm!\n");
+	exit(0);
+}
+
+TAILQ_HEAD(cropq, crop);
+
+struct farm {
+	struct cropq	cropq;
+	struct crop	*cursor;
+};
+
+struct crop {
+	void	(* volatile describe)(struct crop *);
```

Comment:
I would imagine this code is sufficiently obfuscated with regards to UB that the `volatile` is unnecessary?

Reply:
Yea, it results in the same code with and without, so it's gone.